### ISSUE TITLE
Wayland-client won't compile

### ISSUE THREAD

**Issue by Rheydskey** [hecrj/iced#562]
Wayland-client doesn't want to compile, I don't know why.
I tried to install some dependencies but no result.
So I give you the compilation errors in pastebin: https://pastebin.com/ehKUQm2n

**Comment by twitchyliquid64** [comment]
Looking at `wayland-rs` project, [it says](https://github.com/Smithay/wayland-rs#requirements):

> Requires at least rust 1.41 to be used, and version 1.15 of the wayland system libraries <snip>

Can you confirm those are both true?

**Comment by Rheydskey** [comment]
I can deactivate Wayland because I'm on Xorg. Using the features?

**Comment by twitchyliquid64** [comment]
I think the features are all under control of the underlying crates. In any case, the library just implements the protocol, so no biggy if its unused. Im guessing feature flag plumbing is needed through the intermediate crates, or you need to install the library.

If I can put my 'strong-opinion' cap on, I would recommend migrating away from Xorg now that its formally in sunset mode / bitrotting. Wayland is the future!

**Comment by Rheydskey** [comment]
I'd like to use Wayland but the performance in games is too unpleasant to play and DEs like KDE or GNOME are totally finished for Wayland. Also the fact that I have a Nvidia graphics card is a hindrance.
And I looked for the library I have version 1.18 with Arch Linux

**Comment by twitchyliquid64** [comment]
> I'd like to use Wayland but the performance in games is too unpleasant to play and DEs like KDE or GNOME are totally finished for Wayland. Also the fact that I have a Nvidia graphics card is a hindrance.

Ah, Nvidia. AFAICT their approach of bolting on closed-source drivers (via DKMS) instead of participating in kernel development Linux-style results in a ton of bugs and wierdness. I have no idea if their DKMS stuff messed with Wayland.

As far as I can tell anecdotally, it looks like most people who use Linux are avoiding Nvidia hardware due to the constant issues. Personally, I got very frustrated with the bugs I was encountering with Nvidia and switched about a year ago. Everything works perfectly now, and I dont look back XD

> And I looked for the library I have version 1.18 with Arch Linux

This might be an issue with `wayland-rs` - maybe open an issue there?

**Comment by hecrj** [comment]
@Rheydskey I think you may be able to fix the issue by installing the Wayland development libraries.

**Comment by Rheydskey** [comment]
> @Rheydskey I think you may be able to fix the issue by installing the Wayland development libraries.

Do you have the name of the package under ArchLinux ?

**Comment by hecrj** [comment]
Have you tried the [`wayland`](https://www.archlinux.org/packages/extra/x86_64/wayland/) package?

**Comment by Rheydskey** [comment]
Yes, already installed

**Comment by myfreeweb** [comment]
> Compiling wayland-client v0.23.6

That's a suspiciously low version, the latest is [0.28.1](https://github.com/Smithay/wayland-rs/commit/b5617bcf0367e5ee5349e100b2bf24e877bc74af). winit 0.23 does use wayland-rs 0.28. iced [uses winit 0.23 as of 11 days ago](https://github.com/hecrj/iced/pull/542).

…you're not using iced 0.1.1 from crates.io, are you?

**Comment by Rheydskey** [comment]
> …you're not using iced 0.1.1 from crates.io, are you?

I compile each time an example of the github

**Comment by raymanfx** [comment]
FYI I had the same issue on Fedora 33.
Apparently I tried to compile my app against a version of `iced` on GitHub which, at the time, pulled in `wayland-client v0.23.6`. The latest code tracks `winit v0.23` which indeed pulls in `wayland-rs v0.28` (and thus an updated version of `wayland-client` as well).

TL;DR to fix your issue: `cargo update` should be sufficient.

**Comment by Rheydskey** [comment]
Thank you it works

**Comment by myfreeweb** [comment]
Right, heh, I also just saw this on a completely clean checkout of this repo without updates. @hecrj please commit a `cargo update` :)

**Comment by hecrj** [comment]
@myfreeweb [As this is a library](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries), we do not version the `Cargo.lock` file:

https://github.com/hecrj/iced/blob/d0402d072d3f4e128c55fc0a6184c5f7c712bb20/.gitignore#L4
